src/runtimeChoiceSelector.ts

```diff
@@ -0,0 +1,95 @@
+import * as _ from "lodash";
+import * as path from "path";
+import * as clc from "cli-color";
+import * as semver from "semver";
+
+import * as FirebaseError from "./error";
+import * as utils from "./utils";
+
+// have to require this because no @types/cjson available
+// tslint:disable-next-line
+var cjson = require("cjson");
+
+const MESSAGE_FRIENDLY_RUNTIMES: any = {
+  nodejs6: "Node.js 6 (Deprecated)",
+  nodejs8: "Node.js 8",
+  nodejs10: "Node.js 10 (Beta)",
+};
+
+const ENGINE_RUNTIMES: any = {
+  6: "nodejs6",
+  8: "nodejs8",
+  10: "nodejs10",
+};
+
+export const ENGINES_FIELD_REQUIRED_MSG = clc.bold(
+  "Engines field is required in package.json but none was found."
+);
+export const UNSUPPORTED_NODE_VERSION_MSG = clc.bold(
+  `package.json in functions directory has an engines field which is unsupported. ` +
+    `The only valid choices are: ${clc.bold('{"node": "8"}')} and ${clc.bold('{"node": "10"}')}. ` +
+    `Note that Node.js 6 is now deprecated.`
+);
+export const DEPRECATION_WARNING_MSG =
+  clc.bold.yellow("functions: ") +
+  "Deploying functions to Node 6 runtime, which is deprecated. Node 8 is available " +
+  "and is the recommended runtime.";
+
+export const FUNCTIONS_SDK_VERSION_TOO_OLD_WARNING =
```

Comment:
I don't think any of these constant strings should be exported - there's nothing outside this function that would need them, right?

(I see the tests, I'm excluding those in this thought and making a comment down there)

Reply:
(I actually may change my mind on this. I'll leave it up to you - I sent you a internal link out-of-band.)

Reply:
Discussed offline - according to internal unit test best practices documentation:

*Though increasing visibility should be avoided for methods, it's generally fine on constants to help keep tests in sync as those constants change. It can also be valuable on fields representing UI elements, which are visible to the user and part of a class's contract, but not normally exposed via its API.*

I'd like to keep the constants exported, unless there are strong concerns.